program/src/processor.rs

```diff
@@ -206,6 +211,159 @@ impl Processor {
+        mango_group.num_mango_accounts -= 1;
+
+        // Prevent account being loaded by program
+        mango_account.meta_data.is_initialized = false;
+
+        Ok(())
+    }
+
+    #[inline(never)]
+    fn resolve_dust(program_id: &Pubkey, accounts: &[AccountInfo]) -> MangoResult<()> {
+        const NUM_FIXED: usize = 7;
+        let accounts = array_ref![accounts, 0, NUM_FIXED];
+
+        let [
+            mango_group_ai,     // read
+            mango_account_ai,   // write
+            owner_ai,           // read, signer
+            dust_account_ai,    // write
+            root_bank_ai,       // read
+            node_bank_ai,       // write
+            mango_cache_ai      // read
+        ] = accounts;
+
+        let mango_group = MangoGroup::load_checked(mango_group_ai, program_id)?;
+        check_eq!(&mango_account_ai.owner, &program_id, MangoErrorCode::InvalidOwner)?;
+        let mut mango_account =
+            MangoAccount::load_mut_checked(mango_account_ai, program_id, &mango_group_ai.key)?;
+        check_eq!(&mango_account.owner, owner_ai.key, MangoErrorCode::InvalidOwner)?;
+        check!(owner_ai.is_signer, MangoErrorCode::InvalidSignerKey)?;
+
+        let mut dust_account =
+            MangoAccount::load_mut_checked(dust_account_ai, program_id, &mango_group_ai.key)?;
+
+        // Check dust account
+        let (pda_address, _bump_seed) = Pubkey::find_program_address(
+            &[&mango_group_ai.key.as_ref(), b"DustAccount"],
+            program_id,
+        );
+        check!(&pda_address == dust_account_ai.key, MangoErrorCode::InvalidAccount)?;
+        check!(&dust_account.owner == mango_group_ai.key, MangoErrorCode::InvalidOwner)?;
+
+        let mango_cache = MangoCache::load_checked(mango_cache_ai, program_id, &mango_group)?;
+
+        let token_index = mango_group
+            .find_root_bank_index(root_bank_ai.key)
+            .ok_or(throw_err!(MangoErrorCode::InvalidRootBank))?;
+
+        // Find the node_bank pubkey in root_bank, if not found error
+        let root_bank = RootBank::load_checked(root_bank_ai, program_id)?;
+        check!(root_bank.node_banks.contains(node_bank_ai.key), MangoErrorCode::InvalidNodeBank)?;
+        let mut node_bank = NodeBank::load_mut_checked(node_bank_ai, program_id)?;
+
+        // Check validity of root bank cache
+        let now_ts = Clock::get()?.unix_timestamp as u64;
+        let root_bank_cache = &mango_cache.root_bank_cache[token_index];
+        root_bank_cache.check_valid(&mango_group, now_ts)?;
+
+        let borrow_amount = mango_account
+            .get_native_borrow(&mango_cache.root_bank_cache[token_index], token_index)?;
+        let deposit_amount = mango_account
+            .get_native_deposit(&mango_cache.root_bank_cache[token_index], token_index)?;
+
+        // Amount must be dust aka < 1 native spl token
+        if borrow_amount > ZERO_I80F48 && borrow_amount < ONE_I80F48 {
+            checked_change_net(
```

Comment:
use transfer_token_internal when possible. It's a minor point but it makes sure the transfer always works

Reply:
@riordanp